pkg/minikube/logs/logs.go

```diff
@@ -261,13 +261,26 @@ func OutputOffline(lines int, logOutput *os.File) {
 // logCommands returns a list of commands that would be run to receive the anticipated logs
 func logCommands(r cruntime.Manager, bs bootstrapper.Bootstrapper, cfg config.ClusterConfig, length int, follow bool) map[string]string {
 	cmds := bs.LogCommands(cfg, bootstrapper.LogOptions{Lines: length, Follow: follow})
+	containers, err := r.ListContainers(cruntime.ListContainersOptions{})
+	if err != nil {
+		klog.Errorf("Failed to list containers: %v", err)
+		return nil
+	}
```

Comment:
Added a `Name` field under `Container` struct.

This way we only need to do **one** `ListContainers` call during `minikube logs`, and later filter container names in code. This makes `minikube logs` about 15% - 20% faster e2e.

Some benchmark results (done on a minikube profile I was working on with 14 containers):

| `master`     | This branch  |
|--------------|--------------|
| 824.937374ms | 680.089196ms |
| 845.681224ms | 670.916441ms |
| 827.478504ms | 677.877367ms |
| 831.836328ms | 688.230854ms |
| 825.594147ms | 678.318603ms |
| 809.029915ms | 669.383535ms |